src/Vlingo.Xoom/ApplicationProperty.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright © 2012-2020 VLINGO LABS. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Vlingo.Xoom {
+    public class ApplicationProperty {
+        private static string xoomPrefix = "VLINGO_XOOM";
+        private static string combinationPattern = "{0}.{1}";
+
+        public static string ReadValue(string key, Dictionary<string, string> properties) {
+            string propertiesValue = RetrieveFromProperties(key, properties);
+            return propertiesValue != null ? propertiesValue : RetrieveFromEnvironment(key);
+        }
+
+        public static List<string> ReadMultipleValues(string key, string separator, Dictionary<string, string> properties) {
+            string value = ReadValue(key, properties);
+
+            return value == null ? new List<string>() : value.Split(new string[] { separator }, StringSplitOptions.None).ToList();
+        }
+
+        private static string RetrieveFromProperties(string key, Dictionary<string, string> properties) {
+            if (!properties.ContainsKey(key)) {
+                return null;
+            }
+            string value = properties.FirstOrDefault(x => x.Key == key).Value.Trim();
+
+            return string.IsNullOrEmpty(value) ? null : value;
+        }
+
+        private static string RetrieveFromEnvironment(string key) {
+            string envKey = ResolveEnvironmentVariable(key);
+
+            if (SystemHolder.variables.ContainsKey(envKey)) {
+                return null;
```

Comment:
What about returnin empty string rather than null?

Reply:
My concern is about warnings because this is set up in the project `<Nullable>enable</Nullable>`. Actually you can return `null` but you must address warnings if they are. Not sure if there is one for strings if you return null. If there is no it's ok.

Reply:
**but you must address warnings if they are** -> I think you should show me an example about it. Because i didnt get properly

Reply:
Compilation, in the output of your IDE

Reply:
Like that ?
            if (!properties.ContainsKey(key))
            {
#pragma warning disable CS8603 // Possible null reference return.
                return null;
#pragma warning restore CS8603 // Possible null reference return.
            }

Reply:
Is this solution fine?